app/views/compute_resources_vms/form/azurerm/_base.html.erb

```diff
@@ -136,19 +136,15 @@
                :required => true
            }
 %>
+
 <%= text_f f, :password,
            {
                :label    => _('Password'),
-               :required => true
+               :required => true,
+               :help_inline => _('To login to the vm via sshkeys, you can leave password blank and add host paramter called remote_execution_ssh_keys')
            }
 %>
 
-<%= textarea_f f, :ssh_key_data,
```

Comment:
We shouldn't remove the option for supplying external ssh key. What we need is setting up `remote_execution_ssh_keys` parameter automatically with the supplied value.

Reply:
So, as of now, I was testing this by explicitly setting up `remote_execution_ssh_keys` parameter in hosts parameter. Could you please give me some pointers on doing it automatically?

Reply:
@ShimShtein what I feel is we should not set user provided keys with parameter remote_execution_ssh_keys because then ssh keys will be copied when finish template or user data will run successfully. Instead I think better option will be to pass user ssh key at time of creating instance only this will make sure ssh works even if finish or user data does not work.

Reply:
@upadhyeammit, we have discussed it with Aditi. For now fog-azure-rm allows setting only a single ssh key. Since foreman needs to run on a host first, we need to set foremans key with the creation of the template, otherwise, foreman won't be able to run finish template on the host at all.

@apuntamb, since it's a workaround anyway, it would be awesome if we could add a PR to fog-azure-rm to change this limitation.